repose-aggregator/components/services/container-configuration-service/impl/src/main/scala/org/openrepose/nodeservice/containerconfiguration/ContainerConfigurationServiceImpl.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * _=_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=
+ * Repose
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Copyright (C) 2010 - 2015 Rackspace US, Inc.
+ * _-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+package org.openrepose.nodeservice.containerconfiguration
+
+import java.lang.Long
+import java.util.Optional
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicReference}
+import java.util.function.UnaryOperator
+import javax.annotation.{PostConstruct, PreDestroy}
+import javax.inject.{Inject, Named}
+
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import org.openrepose.commons.config.manager.UpdateListener
+import org.openrepose.core.container.config.{ContainerConfiguration, DeploymentConfiguration}
+import org.openrepose.core.services.config.ConfigurationService
+import org.openrepose.core.spring.ReposeSpringProperties
+import org.springframework.beans.factory.annotation.Value
+
+import scala.collection.JavaConversions._
+
+@Named
+class ContainerConfigurationServiceImpl @Inject()(@Value(ReposeSpringProperties.NODE.CLUSTER_ID) clusterId: String,
+                                                  configurationService: ConfigurationService)
+  extends ContainerConfigurationService with UpdateListener[ContainerConfiguration] with LazyLogging {
+
+  import ContainerConfigurationServiceImpl._
```

Comment:
I'm actually not sure. That's an interesting question. I wonder if that doesn't work since the companion object is in the same file, and uses the same file scope imports.

Let me try it and see what happens.